### DISCUSSION TITLE
json: filtering request based on time and Dashboard variables

### DISCUSSION THREAD

**Discussion by nickgiant** [grafana/grafana-infinity-datasource#759] [Q&A] [answered]
Hi
Infinity is a very interesting plugin for Grafana.
I was wondering if there is a way to filter data based on time period selected, plus dashboard variables selected. I already have implement it in the server side (with url parameters), but I would like to know if there is a way to map the ui selection, with the url parameters (or any other way).
Keep up the good work.

**Comment by nickgiant** [comment]
Looking around I found this answer about a dashboard variable 
[https://github.com/grafana/grafana-infinity-datasource/discussions/655#discussioncomment-6854340](https://github.com/grafana/grafana-infinity-datasource/discussions/655#discussioncomment-6854340)
But what about filtering the time range with time components show by default in grafana?

**Reply by yesoreyeram** [reply]
You can use grafana time variables in any part of your query. They will be interpolated into actual time ranges in grafana. You can pass them as appropriate parameters required by your API

**Reply by yesoreyeram** [reply, marked as answer]
https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#__from-and-__to
